network-store-iidm-impl/src/main/java/com/powsybl/network/store/iidm/impl/TerminalImpl.java

```diff
@@ -306,46 +346,90 @@ boolean identifySwitchToOpenPath(GraphPath<Integer, Edge> path, Predicate<? supe
         return false;
     }
 
-    private boolean disconnectNodeBreaker(Resource<VoltageLevelAttributes> voltageLevelResource, Predicate<Switch> isSwitchOpenable) {
+    private boolean isAnOpenSwitch(Edge edge) {
+        if (edge.getBiConnectable() instanceof SwitchAttributes switchAttributes) {
+            // Get the switch behind the switchAttributes
+            Optional<SwitchImpl> sw = index.getSwitch(switchAttributes.getResource().getId());
+
+            // Test if the switch is already open
+            return sw.isPresent() && SwitchPredicates.IS_OPEN.test(sw.get());
+        }
+        return false;
+    }
+
+    boolean getDisconnectingSwitches(Predicate<Switch> isSwitchOpenable, Set<SwitchImpl> switchesToOpen) {
+
+        // Voltage level
+        Resource<VoltageLevelAttributes> voltageLevelResource = getVoltageLevelResource();
+
         // Full graph of the network
-        Graph<Integer, Edge> graph = NodeBreakerTopology.INSTANCE.buildGraph(index, voltageLevelResource, true, true);
+        JGraphTGraph graph = new JGraphTGraph(NodeBreakerTopology.INSTANCE.buildGraph(index, voltageLevelResource, true, true));
 
         // Node of the present terminal (start of the paths)
         int node = getAttributes().getNode();
 
         // Nodes of the busbar sections (end of the paths)
         Set<Integer> busbarSectionNodes = getBusbarSectionNodes(voltageLevelResource);
 
-        // Find all paths from the source to the targets
-        AllDirectedPaths<Integer, Edge> allDirectedPaths = new AllDirectedPaths<>(graph);
-        List<GraphPath<Integer, Edge>> allPaths = allDirectedPaths.getAllPaths(Set.of(node), busbarSectionNodes, true, null);
```

Comment:
instead of rewriting our own BFS, is it possible to use https://jgrapht.org/javadoc/org.jgrapht.core/org/jgrapht/traverse/BreadthFirstIterator.html ?

Reply:
(or any other method I don't know the jgrapht api)